spec/requests/api/v2/key_performance_indicators_controller_spec.rb

```diff
@@ -0,0 +1,473 @@
+# frozen_string_literal: true
```

Comment:
See my comments above on breaking this up. Most of these tests can be moved to model tests, which are much simpler and focused on the individual testable unit.

Reply:
We should still have a KPI request spec, but that should just be asserting that the API behaves in a way consistent with its contract: 404 for indicators that don't exist, authorization errors, etc.